test/functional/feature_chain_tiebreaks.py

```diff
@@ -99,5 +99,53 @@ def run_test(self):
         # B7 is now active.
         assert_equal(node.getbestblockhash(), blocks[7].hash)
 
+        # Invalidate blocks to start fresh on the next test
+        node.invalidateblock(blocks[0].hash)
+
+    def test_chain_split_from_disk(self):
+        node = self.nodes[0]
+        peer = node.add_p2p_connection(P2PDataStore())
+
+        self.log.info('Precomputing blocks')
+        #
+        #      A1
+        #     /
+        #   G
+        #     \
+        #      A2
+        #
+        blocks = []
+
+        # Construct two blocks building from genesis.
+        start_height = node.getblockcount()
+        genesis_block = node.getblock(node.getblockhash(start_height))
+        prev_time = genesis_block["time"]
+
+        for i in range(0, 2):
+            blocks.append(create_block(
+                hashprev=int(genesis_block["hash"], 16),
+                tmpl={"height": start_height + 1,
+                # Make sure each block has a different hash.
+                "curtime": prev_time + i + 1,
+                }
+            ))
+            blocks[-1].solve()
+
+        # Send blocks and test the last one is not connected
+        self.log.info('Send A1 and A2. Make sure than only the former connects')
+        peer.send_blocks_and_test([blocks[0]], node, success=True)
+        peer.send_blocks_and_test([blocks[1]], node, success=False)
+
+        self.log.info('Restart the node and check that the best tip before restarting matched the ones afterwards')
+        # Restart and check enough times to this to eventually fail if the logic is broken
+        for _ in range(10):
+            self.restart_node(0)
+            assert_equal(blocks[0].hash, node.getbestblockhash())
```

Comment:
In c6ca2a17ea4d76f21d7702f70ba892a9494576bf "test: Adds block tiebreak over restarts tests"

This does not seem to fail on master, but it does on this branch with the fix commit reverted. Is it possible that this was fixed by a different PR?

Reply:
for me, it also fails on master - but not always, intermittently.